app/services/carrierwave_filename.rb

```diff
@@ -2,7 +2,7 @@ module CarrierwaveFilename
   def for(attachment, underscore: false)
     file = attachment.file && attachment.file.path.split('/').last
 
-    underscore ? file.tr('-', '_') : file
+    underscore ? file.gsub(/[^a-zA-Z0-9](?=[[:alnum:]]*\.)/, '_') : file
```

Comment:
Sanitisation could be hoisted into a private method with a descriptive name. It's no longer immediately clear what is happening here due to the relative complexity of the regex. Do you think it would help?